.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+repos:
+  - repo: https://github.com/ambv/black
+    rev: 22.3.0
+    hooks:
+    - id: black
+  - repo: https://github.com/nbQA-dev/nbQA
+    rev: 1.2.3
+    hooks:
+    - id: nbqa-black
+    - id: nbqa-isort
+    - id: nbqa-flake8
```

Comment:
Hmm, maybe we switch to [ruff](https://github.com/astral-sh/ruff) instead of flake8, it's much faster and meshes fine with Diffrax.